src/shippingservice/src/telemetry/logs_conf.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use opentelemetry_appender_tracing::layer::OpenTelemetryTracingBridge;
+use opentelemetry_sdk::{logs::Config, runtime};
+
+use tracing_subscriber::prelude::*;
+
+use opentelemetry_sdk::logs as sdklogs;
+
+use super::get_resource_attr;
+
+pub fn init_logger() -> Result<sdklogs::Logger, opentelemetry::logs::LogError> {
+    opentelemetry_otlp::new_pipeline()
+        .logging()
+        .with_log_config(Config::default().with_resource(get_resource_attr()))
+        .with_exporter(opentelemetry_otlp::new_exporter().tonic())
+        .install_batch(runtime::Tokio)
+}
+
+pub fn configure_global_logger(tracer: opentelemetry_sdk::trace::Tracer) {
+    let telemetry = tracing_opentelemetry::layer().with_tracer(tracer);
+    let logger_provider = opentelemetry::global::logger_provider();
+    let layer = OpenTelemetryTracingBridge::new(&logger_provider);
+    let _ = tracing_subscriber::registry()
+        .with(layer)
+        .with(telemetry)
```

Comment:
what are we using `tracing_opentelemetry` crate for? From what I can see, we are using the otel tracing API for traces, and the tokio-tracing api (through OpenTelemetryTracingBridge) for logs.

Reply:
@lalitb I'm not a Rust expert, I did a bunch of try and fail in order to land in this.
What I can say is that without the `telemetry` layer I wasn't getting the `POST` request span, which comes from the `reqwest` instrumentation, and the trace was broken.
Once I've added it, everything was back to properly working.

Reply:
Not fully familiar with how tracing-opentelemetry works, but my **guess** of how this is attempted to be setup in this PR is:
1. Uses tokio-tracing as the *logging* API in the shipping service.
2. Uses OpenTelemetryTracingBridge, to send the logs from above to otel->otlp.
3. Also enables `reqwest-tracing` crate, which (I am **guessing** here..), is using tokio-tracing for *tracing* needs (i.e to create Spans, and possibly SpanEvents (in OTel terminology) or just Events in tokio-tracing terminology)
4. The `tracing_opentelemetry` layer is to send the traces from above to otel->otlp.

Good chance there will be duplication of telemetry if `SpanEvents` are used, as (2) will convert them to OTLP Logs, and (4) will convert them to SpanEvents inside OTLP Traces.

Unfortunately, I don't have a good suggestion to fix this, at least until https://github.com/open-telemetry/opentelemetry-rust/issues/1571 is resolved.
I understand it is frustrating, and the same frustration is faced by end users as well, which is why OTel Rust community is trying to address this and not declare 1.0/GA until this is solved.

Reply:
It seems to me that your guess is correct, @cijothomas. And good point on duplicate events (as SpanEvents and Logs) Unfortunately, this can't be avoided if `reqwest-tracing` needs `tracing-opentelemetry`.

Reply:
Hmmmm... Okay?

Should I just forget the logs, and only send the PR fixing the Context Prop?

Reply:
Give us ~1 more week. Will discuss with OTel Rust community and get back to you on how best to resolve this!

Reply:
From OTel Rust SIG/community meeting:
1. Please scope down this PR, to only address the context propagation issue.
2. Remove tracing and tracing-appender, as that will cause duplicate telemetry via SpanEvent and LogRecord.
3. Come back and add logging via tracing+tracing-appender, once OTel Rust fixes the underlying issue.

Reply:
@cijothomas thanks for keeping me in the loop!

I've removed the tracing-appender for logging, but I can't remove tracing as it is used in the `reqwest_tracing`.
I've updated the PR and in theory it should be good to go.

Would you be able to double check?